server/trace-func-go/aws/trace_func.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"time"
+)
+
+// static double SQRTSD (double x) {
+//     double r;
+//     __asm__ ("sqrtsd %1, %0" : "=x" (r) : "x" (x));
+//     return r;
+// }
+import "C"
+
+const (
+	// ContainerImageSizeMB was chosen as a median of the container physical memory usage.
+	// Allocate this much less memory inside the actual function.
+	ContainerImageSizeMB = 15
+)
+
+const EXEC_UNIT int = 1e2
+
+var iterationsMultiplier = 102
+
+// Response is of type APIGatewayProxyResponse since we're leveraging the
+// AWS Lambda Proxy Request functionality (default behavior)
+//
+// https://serverless.com/framework/docs/providers/aws/events/apigateway/#lambda-proxy-integration
+type Response events.APIGatewayProxyResponse
+
+func takeSqrts() C.double {
+	var tmp C.double // Circumvent compiler optimizations
+	for i := 0; i < EXEC_UNIT; i++ {
+		tmp = C.SQRTSD(C.double(10))
+	}
+	return tmp
+}
+
+func busySpin(runtimeMilli uint32) {
+	totalIterations := iterationsMultiplier * int(runtimeMilli)
+
+	for i := 0; i < totalIterations; i++ {
+		takeSqrts()
```

Comment:
Have you tried to reuse the code from the `pkg/workload`? Since you are compiling it that shouldn't be a problem to import this module as it is done in `server/trace-func-go/trace_func.go`.